### PR TITLE
JESI-3014

### PR DESCRIPTION
**Pull request by AndreTheHunter**
[JESI-3014](https://jesims.atlassian.net/browse/JESI-3014)
- [x] https://github.com/jesims/parent/pull/38
- [x] https://github.com/jesims/build-bus/pull/27
- [ ] https://github.com/jesims/customs/pull/2

### CODE REVIEW

File: test/io/jesi/backpack/atom_test.cljc [hunk 1]
```diff
@@ -97,3 +97,99 @@
...
+          [:something :else] "value"
+          :range (range 0 5)
+          [:meaning :of :life] 42)
+        (is-equal? {:this      :that
+                    :range     [0 1 2 3 4]
+                    :something {:else "value"}
+                    :meaning   {:of {:life 42}}})
+        (is-called-once?)))))
+
+
```
Comment (axrs) [inline test/io/jesi/backpack/atom_test.cljc]: 👾

File: src/io/jesi/backpack/random.cljc [hunk 2]
```diff
@@ -55,7 +55,7 @@
     (apply str)))
 
 (defn- gen-str [chars size]
-  (apply str (take size (repeatedly #(rand-nth chars)))))
+  (clojure.string/join (repeatedly size (fn* [] (rand-nth chars)))))
```
Comment (BadAlgorithm) [inline src/io/jesi/backpack/random.cljc]: Why use special form `fn*`?
(I'm getting deja vu, I think I may have asked this question in the past but forgot the rationale)

Comment (axrs) [inline src/io/jesi/backpack/random.cljc]:
```suggestion
  (str/join (repeatedly size (fn* [] (rand-nth chars)))))
```
import as str

What's the significance of `fn*`?

File: .clj-kondo/config.edn [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ :lint-as {io.jesi.backpack.macros/def-        clojure.core/def
+           io.jesi.customs.strict/are          clojure.test/are
+           io.jesi.customs.strict/deftest      clojure.test/deftest
+           io.jesi.customs.strict/is           clojure.test/is
+           io.jesi.customs.strict/testing      clojure.test/testing
+           io.jesi.customs.strict/use-fixtures clojure.test/use-fixtures
+           io.jesi.backpack.macros/reify-ifn   clojure.core/reify}
+ :linters {:unresolved-symbol {:level :off}
+           :refer-all         {:level :off}
+           :unused-binding    {:level :off}}}
```
[the remark this answers is not in the document]
Comment (AndreTheHunter) [reply .clj-kondo/config.edn]: Yup. Updated bindle kondo config.

File: src/io/jesi/backpack/random.cljc (hunk 2 above)
Comment (AndreTheHunter) [reply src/io/jesi/backpack/random.cljc]: I think that was kibit. I'll revert

File: test/io/jesi/backpack/fn_test.cljc [hunk 4]
```diff
@@ -111,7 +111,7 @@
     (testing "is partial ="
       (let [=1 (partial = 1)]
         (is= (=1 1)
-             ((bp/p= 1) 1)))
+          ((bp/p= 1) 1)))
```
Comment (axrs) [inline test/io/jesi/backpack/fn_test.cljc]: formatter has messed up

File: test/io/jesi/backpack/fn_test.cljc [hunk 5]
```diff
@@ -137,7 +137,7 @@
   (testing "and-fn"
 
     (testing "throws an exception if no parameters are given"
-      (is (thrown? #?(:clj ArityException :cljs js/Error) (bp/and-fn))))
+      (is (thrown? #?(:clj ArityException :cljs js/Error) (eval `(bp/and-fn)))))
```
Comment (axrs) [inline test/io/jesi/backpack/fn_test.cljc]: why eval?

File: src/io/jesi/backpack/random.cljc (hunk 2 above)
Comment (AndreTheHunter) [reply src/io/jesi/backpack/random.cljc]: Removed. That was kibit

File: test/io/jesi/backpack/fn_test.cljc (hunk 5 above)
Comment (AndreTheHunter) [reply test/io/jesi/backpack/fn_test.cljc]: Lint (like eastwood) does not like calling the function with incorrect arity. This way we can still test the behaviour but without the linter crying.

**Pull request merged**